scripts/rvnbag_play.py

```diff
@@ -271,37 +229,36 @@ def __init__( self, bagfile, loop):
         rospy.loginfo("...Done")
 
         # create clock publisher
-        self._clock_publisher = rospy.Publisher("/clock", Clock, queue_size=100)
+        self._clock_publisher = rospy.Publisher("/clock", Clock, queue_size=1, tcp_nodelay=True)
 
 
-    def _create_publisher( self, msg_cont ):
+    def _create_publisher( self, topic, msg_class ):
         try:
             new_pub = rospy.Publisher(
-                msg_cont.topic,
-                msg_cont.msg_class,
-                latch      = self.bag_reader.is_latched(msg_cont.topic),
-                queue_size = 1000 ) # RVN::FIX: queue_size=???
-
+                topic,
+                msg_class,
+                latch      = self.bag_reader.is_latched(topic),
+                queue_size = 16,
+                tcp_nodelay = True,
```

Comment:
Another obscure call-out here. This disables a network efficiency optimization when sending ros messages on the wire, which has the downside introducing more variable delay to message publishing.

Reply:
By disabling it, we're trading decreased network/bandwidth efficiency for lower, less variable delays when sending messages.

Reply:
If `tcp_nodelay` is left `False`, then [Nagle's algorithm](https://searchnetworking.techtarget.com/definition/Nagles-algorithm) is used to batch smaller message buffers together in a single TCP tx buffer ( and possibly as well on the rx buffer on the rviz side, would have to read up more)